File: alarmclock/src/test/java/com/externalalarmclock/alarmclock/resources/SetNextAlarmResourceTest.java

package com.externalalarmclock.alarmclock.resources;

import com.externalalarmclock.alarmclock.AlarmStore;
import com.externalalarmclock.lib.rpiws281x.RpiWs281xChannel;
import com.externalalarmclock.pojo.ExternalAlarm;

import java.util.List;
import java.util.Arrays;

import org.junit.Test;
import org.junit.Assert;

public class SetNextAlarmResourceTest {
	private final AlarmStore alarmStore = new AlarmStore();
	private final SetNextAlarmResource resource = new SetNextAlarmResource(alarmStore);
	private final RpiWs281xChannel channel = new RpiWs281xChannel();
	@Test
	public void setNextAlarmTest() {
		alarmStore.addChannels(Arrays.asList(channel));
		ExternalAlarm alarm = new ExternalAlarm();
		resource.setNextAlarm(alarm);

		Assert.assertNull(alarmStore.getNextAlarm(channel));
	}
}

File: alarmclock/src/test/kotlin/SetNextAlarmResourceTest.kt
package com.externalalarmclock.alarmclock.resources

import com.externalalarmclock.alarmclock.AlarmStore
import com.externalalarmclock.lib.rpiws281x.RpiWs281xChannel
import com.externalalarmclock.pojo.ExternalAlarm
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

import java.time.Duration

class SetNextAlarmResourceTest {
    private val alarmStore = AlarmStore()
    private val resource = SetNextAlarmResource(alarmStore)
    private val channel = RpiWs281xChannel()
    @Test
    fun setNextAlarmTest() {
        alarmStore.wakeUpLightDuration = Duration.ofMinutes(30)
        alarmStore.addChannels(listOf(channel))
        val alarm = ExternalAlarm()
        resource.setNextAlarm(alarm)

        Assertions.assertEquals(null to null, alarmStore.getNextAlarm(channel))
    }
}
